ARCs/arc-0035.md

```diff
@@ -0,0 +1,119 @@
+---
+arc: 35
+title: Algorand Offline Wallet Backup Protocol
+description: Wallet-agnostic backup protocol for multiple accounts
+author: Yigit Guler (@yigitguler)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/156
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-01-03
+---
+# Algorand Secure Offline Wallet Backup Protocol
+
+
+## Abstract
+This document outlines the high-level requirements for a wallet-agnostic backup protocol that can be used across all wallets on the Algorand ecosystem.
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+### Requirements
+At a high-level, offline wallet backup protocol has the following requirements:
+
+ * Wallet applications should allow backing up and storing multiple accounts at the same time.
+Account information should be encrypted with a user defined secret key, utilizing NaCl SecretBox method (audited and endorsed by Algorand).
+ 
+ * Encrypted final string should be easily copyable to be stored digitally.
+When importing, wallet applications should be able to detect already imported accounts and gracefully ignore them.
+
+### Format
+
+Before encryption, account information should be converted to the following JSON format:
+
+```
+{
+  "device_id": "UNIQUE IDENTIFIER FOR DEVICE (OPTIONAL)",
+  "provider_name": "PROVIDER NAME (OPTIONAL, i.e. Pera Wallet)",
+  "accounts": [
+    {
+      "address": "ACCOUNT PUBLIC ADDRESS (REQUIRED)"
+      "name": "USER DEFINED ACCOUNT NAME (OPTIONAL)",
+      "account_type": "TYPE OF ACCOUNT: single, multisig, watch, contact, ledger (REQUIRED)",
+    "private_key": "ACCOUNT PRIVATE KEY AS BASE64 ENCODING OF 64 BYTE PRIVATE KEY as encoded by NACL/libsodium (NOT PASSPHRASE, REQUIRED for user-owned accounts, can be omitted in case of watch, contact, ledger accounts)"
```

Comment:
I wanted to confirm one thing regarding multisig accounts, in the JSON format for the private key, it seems to me that it would be a base 64 encoding of the object that has its data, for example:
```
{
        version: 1,
        threshold: 2,
        addrs: [
            account1.addr,
            account2.addr,
            account3.addr,
        ],
};
```
Is this correct?

Reply:
@yuli-ferna, this is a great point. Apparently, we will need a new optional field: multi_sig. And it can contain this format. How does it sound?

Reply:
Formatting in base64 will complicate the debugger's mind for no reason. This way, it can be easier to read & debug.

Reply:
I updated it. Please review if it looks OK.

Reply:
Perfect! Thank you :)